### DISCUSSION TITLE
novelWriter also works on the Haiku Operating System

### DISCUSSION THREAD

**Discussion by ghost** [vkbo/novelWriter#2050] [General]
Good day,

Just let you know that novelWriter (tested 2.5.1, 2.5.2 and 2.6.a1) work on Haiku, the Operating System that inherits BeOS ideas.

We are working on a package to make it available to all Haiku users and soon will be installable.

Regards,
RR

P.S.: Would it be possible to name the .md files with a the name assigned inside the software?

**Comment by vkbo** [comment]
That's great! Thanks for the info.

> P.S.: Would it be possible to name the .md files with a the name assigned inside the software?

Sorry, no. The files are named that way so that the names are static throughout their lifetime, the software doesn't need to consider character encoding and legal/illegal characters, and last but not least, discourage editing.

They are also not actually md files. Although I'm considering switching the extension to .txt.

I do plan to add a feature to export the project as text/md files mirroring the internal project structure though. See #1108.

**Comment by johnblommers** [comment]
> Just let you know that novelWriter (tested 2.5.1, 2.5.2 and 2.6.a1) work on Haiku, the Operating System that inherits BeOS ideas.
> 
> We are working on a package to make it available to all Haiku users and soon will be installable.

Thanks for this. I've been following Haiku for years and find it to be a very lean, mean, snappy and stable operating system.

**Comment by ghost** [comment]
This is the proof, novelWriter running on Haiku
![novelWriter5](https://github.com/user-attachments/assets/37755317-4d27-43e5-96ef-955858792a4d)
:

![novelWriter4](https://github.com/user-attachments/assets/7fa72a57-246a-4c85-a683-7ae4ab059e28)

> Thanks for this. I've been following Haiku for years and find it to be a very lean, mean, snappy and stable operating system.

No need to thank, I plan to use it myself ;)

> They are also not actually md files. Although I'm considering switching the extension to .txt.
> 
> I do plan to add a feature to export the project as text/md files mirroring the internal project structure though. See #1108.


Yes, I've noticed that they are not really .md files :) 
The option to export to md keeping the folder structure would be nice too... Though I presume I need to get a deeper understanding on novelWriter, and that will come after usage.  I've grown fond of .md lately, so I would vote for .md export ;)

When the package is available to install will post here. 

Thanks and regards,
RR

**Reply by vkbo** [reply]
You can of course generate markdown files from the Manuscript tool. Both standard and extended syntax. But there they are combined into a single manuscript document, so you don't preserve the internal structure.

The idea for the export is to generate individual files in any of the available formats, which includes valid Markdown too.
